### PR TITLE
Fix bug where suggestions could never be set if falsy

### PR DESCRIPTION
**Pull request by josephschmitt**
Also added a ton of missing tests to cover the updateSuggestions method and what it does with the source function.

### CODE REVIEW

File: src/angularComponent/ngcOmniboxController.js [hunk 1]
```diff
@@ -430,12 +434,56 @@ export default class NgcOmniboxController {
...
+   */
+  updateSuggestions() {
+    this._suggestionsUiList.length = 0;
+    this.hint = null;
+
+    this.highlightedIndex = -1; // Forcibly select nothing
+    this._showLoading();
+    this.hideSuggestions = false;
+
+    const promise = this.source({query: this.query, suggestions: this.suggestions});
```
Comment (drawyan) [inline src/angularComponent/ngcOmniboxController.js]: I think this will be cleaner:
this.source({query, suggestion} = this);

Comment (josephschmitt) [inline src/angularComponent/ngcOmniboxController.js]: Oh man, I always forget you can do destructuring on that side too

Comment (josephschmitt) [inline src/angularComponent/ngcOmniboxController.js]: Hmm, linter's not happy about it

File: README.md [hunk 2]
```diff
@@ -163,7 +163,14 @@ displayed.
 To display a hint to the user in addition to suggestions, resolve the promise with an object that
 has keys for `hint` and `suggestions`: `{hint: 'My hint', suggestions: [...]}`. A hint is displayed
 to the right of the text that has been input by the user. Pressing RIGHT on the keyboard replaces
-the input text query with the hint.
+the input text query with the hint. The hint should include the entirety of the query, plus whatever
+else you want to hint with. When displayed, only the addition to the query is shown as a hint, but
+if the using presses RIGHT to complete it, it will use the entirety of the hint provided. For
```
Comment (onlywei) [inline README.md]: if the using?

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js [hunk 3]
```diff
@@ -18,6 +18,55 @@ describe('ngcOmnibox.angularComponent.ngcOmniboxController', () => {
...
 
+  describe('when populating suggestions via the source function', () => {
+    it('it should empty out the suggestions when resolving to a falsy value', (done) => {
+      omniboxController.query = 'my query';
+      omniboxController.suggestions = ['suggestion 1', 'suggestion 2'];
+      omniboxController.source = () => Promise.resolve(null);
+      omniboxController.updateSuggestions().then(() => {
+        expect(omniboxController.suggestions).toBe(null);
+        done();
+      });
```
Comment (onlywei) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: use this format:

```
makeSomePromise().then(() => {
  // make some expect() calls
}).then(done, fail);
```

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js [hunk 4]
```diff
@@ -18,6 +18,55 @@ describe('ngcOmnibox.angularComponent.ngcOmniboxController', () => {
...
+        expect(omniboxController.suggestions.join(',')).toBe('new suggestion 1,new suggestion 2');
+        done();
+      });
+    });
+
+    it('it should throw an error when resolving to a truthy value that\'s not an Array', (done) => {
+      omniboxController.query = 'my query';
+      omniboxController.suggestions = ['suggestion 1', 'suggestion 2'];
+      omniboxController.source = () => Promise.resolve(true);
+      omniboxController.updateSuggestions().catch((e) => {
```
Comment (onlywei) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: Try this format:

```
makeSomePromise.then(fail, (err) => {
  expect()
}).then(done, fail);
```

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js (hunk 3 above)
Comment (josephschmitt) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: what's the advantage?

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js (hunk 4 above)
Comment (josephschmitt) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: Isn't the 2nd param the "old way" and the new way is a catch block?

File: README.md (hunk 2 above)
Comment (josephschmitt) [inline README.md]: Fixed

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js (hunk 4 above)
Comment (josephschmitt) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: This is also harder for me to reason what's going on

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js (hunk 3 above)
Comment (josephschmitt) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: Done

File: spec/tests/angularComponent/ngcOmniboxControllerSpec.js (hunk 4 above)
Comment (josephschmitt) [inline spec/tests/angularComponent/ngcOmniboxControllerSpec.js]: Done

**Pull request merged**